fire/SFMainMod.F90

```diff
@@ -835,6 +1034,275 @@ subroutine  area_burnt_intensity ( currentSite, bc_in )
+             beta_op = 0.200395_r8 *(fuel_sav**(-0.8189_r8))
+             beta_ratio = beta/beta_op  
+
+             ! -- heat of pre-ignition --
+             q_ig = q_dry + 2594.0_r8 * fuel_eff_moist
+
+             ! ---effective heating number---
+             ! Eq A3 in Thonicke et al. 2010.  
+             eps = exp(-4.528_r8 / fuel_sav)     
+             ! Eq A7 in Thonicke et al. 2010 per Eq 49, Rothermel 1972
+             b = 0.15988_r8 * (fuel_sav**0.54_r8)
+             ! Eq A8 in Thonicke et al. 2010 per Eq 48, Rothermel 1972 
+             c = 7.47_r8 * (exp(-0.8711_r8 * (fuel_sav**0.55_r8))) 
+             ! Eq A9 in Thonicke et al. 2010. (typo in Eq A9, using coefficient Eq 50, Rothermel 1972)
+             e = 0.715_r8 * (exp(-0.01094_r8 * fuel_sav))
+
+             midflame_wind = currentSite%wind *0.40_r8  !Scott & Reinhardt 2001 40% open wind speed
+
+             ! Eq A5 in Thonicke et al. 2010
+             ! convert current_wspeed (wind at elev relevant to fire) from m/min to ft/min for Rothermel ROS eqn
+             phi_wind = c * ((3.281_r8*midflame_wind)**b)*(beta_ratio**(-e)) !unitless
+
+             ! ---propagating flux = xi (unitless) 
+             ! Eq A2 in Thonicke et al.2010 and Eq 42 Rothermel 1972
+             xi = (exp((0.792_r8 + 3.7597_r8 * (fuel_sav**0.5_r8)) * (beta+0.1_r8))) / &
+                (192_r8+7.9095_r8 * fuel_sav) 
+
+             ! ---reaction intensity----
+             ! Eq in table A1 Thonicke et al. 2010. 
+             a = 8.9033_r8 * (fuel_sav**(-0.7913_r8))
+             a_beta = exp(a*(1.0_r8-beta_ratio))  !dummy variable for reaction_v_opt equation
+  
+             ! Eq in table A1 Thonicke et al. 2010.
+             ! reaction_v_max and reaction_v_opt = reaction velocity in units of per min
+             ! reaction_v_max = Eq 36 in Rothermel 1972 and Fig 12 
+             reaction_v_max  = 1.0_r8 / (0.0591_r8 + 2.926_r8* (fuel_sav**(-1.5_r8)))
+             ! reaction_v_opt =  Eq 38 in Rothermel 1972 and Fig 11
+             reaction_v_opt = reaction_v_max*(beta_ratio**a)*a_beta
+
+             ! mw_weight = relative fuel moisture/fuel moisture of extinction
+             mw_weight = fuel_eff_moist/fuel_mef
+       
+             ! Eq in table A1 Thonicke et al. 2010. (unitless)
+             moist_damp = max(0.0_r8,(1.0_r8 - (2.59_r8 * mw_weight) + (5.11_r8 * (mw_weight**2.0_r8)) - &
+               (3.52_r8*(mw_weight**3.0_r8))))
+
+             ! ir = reaction intenisty in kJ/m2/min
+             ! sum_fuel as kgBiomass/m2 for ir calculation
+             ir = reaction_v_opt*(net_fuel)*SF_val_fuel_energy*moist_damp*SF_val_miner_damp  
+ 
+             ! actual ROS (m/min) for FM 10 fuels for open windspeed, Eq 8 Scott & Reinhardt 2001
+             ROS_active = 3.34_r8 * ((ir*xi*(1.0_r8+phi_wind)) / (fuel_bd * eps * q_ig))
+
+             ! critical min rate of spread (m/min) for active crowning
+             ROS_active_min = (critical_mass_flow_rate / fuel_bd) * 60.0_r8
```

Comment:
@jkshuman is this using the correct equation?  To me it looks like it's supposed to be calculating the critical minimum rate of spread for active crowning, which is Eq 14 in Scott & Reinhardt.  But this line corresponds to  Eq 12, which is for calculating the rate of spread for initiation.

Reply:
Sorry @jkshuman I see I was getting variable names mixed up and that ROS_active_min is the Scott and Reinhardt R'initiation.  You can ignore my previous comment.